src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountSupervisor.kt

```diff
@@ -1544,46 +1550,28 @@ internal class SubaccountSupervisor(
                 destinationSubaccountNumber = subaccountNumber,
             )
 
-            val transferPayloadString = Json.encodeToString(transferPayload)
-            transferPayloadStrings.add(transferPayloadString)
+            if (childSubaccountNumber !in reclaimingChildSubaccountNumbers) {
+                reclaimingChildSubaccountNumbers.add(childSubaccountNumber)
+                transferPayloads.add(transferPayload)
+            }
         }
 
-        recursivelyReclaimChildSubaccountFunds(transferPayloadStrings)
-    }
-
-    private fun recursivelyReclaimChildSubaccountFunds(transferPayloadStrings: MutableList<String>) {
-        if (transferPayloadStrings.isNotEmpty()) {
-            val transferPayloadString = transferPayloadStrings.removeAt(0)
-            helper.transaction(TransactionType.SubaccountTransfer, transferPayloadString) { response ->
+        transferPayloads.forEach { transferPayload ->
+            val transferPayloadString = Json.encodeToString(transferPayload)
+            val transactionCallback = { response: String? ->
+                this.reclaimingChildSubaccountNumbers.remove(transferPayload.subaccountNumber)
```

Comment:
do we want to remove from set even if there was an error?

Reply:
yea i think so, will rather let it retry the transfer?

Reply:
chatted offline, not retrying makes sense here - unclear if a retry would help the problem. let's add some error logging (maybe amplitude?) so we can track this